base/set.jl

```diff
@@ -621,37 +622,26 @@ julia> replace!(isodd, A, 0, count=2)
 ```
 """
 replace!(pred::Callable, A, new; count::Integer=typemax(Int)) =
-    replace!(x -> if pred(x) Some(new) end, A, count=count)
+    replace!(x -> ifelse(pred(x), new, x), A, count=count)
 
 """
-    replace!(prednew::Function, A; [count::Integer])
+    replace!(new::Function, A; [count::Integer])
 
-For each value `x` in `A`, `prednew(x)` is called and must
-return either `nothing`, in which case no replacement occurs,
-or a value, possibly wrapped as a [`Some`](@ref) object, which
-will be used as a replacement for `x`.
+Replace each element `x` in collection `A` by `new(x)`.
+If `count` is specified, then replace at most `count` values in total
+(when `new(x) === x`, this does not count as a replacement).
```

Comment:
Maybe rephrase this as something like "(replacements being defined as `new(x) !== x`)"?